electrum/commands.py

```diff
@@ -1159,8 +1159,10 @@ async def decode_invoice(self, invoice: str):
         invoice = Invoice.from_bech32(invoice)
         return invoice.to_debug_json()
 
-    @command('wnl')
-    async def lnpay(self, invoice, timeout=120, wallet: Abstract_Wallet = None):
+    @command('wnpl')
+    async def lnpay(self, invoice, timeout=120, password=None, wallet: Abstract_Wallet = None):
+        if wallet.has_password():
+            wallet.check_password(password)
```

Comment:
Conceptually this should not be done here I think.
I moved this check to the `@command` decorator in https://github.com/spesmilo/electrum/commit/37d090c621006cc4caa7e0fb9d080b3e6a9da05c.

Reply:
agreed